### PR TITLE
add special handling for exceptions in JSON format

### PR DESCRIPTION
**Pull request by ericphanson**
based on https://github.com/beacon-biosignals/BeaconK8sUtilities.jl/pull/14

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,7 +37,18 @@ julia> with_logger(FormatLogger(LoggingFormats.JSON(; recursive=true), stderr))
...
..._user_input(ast::Any, backend::REPL.REPLBackend)\n    @ REPL ~/.asdf/installs/julia/1.7.0/share/julia/stdlib/v1.7/REPL/src/REPL.jl:150\n  [7] repl_backend_loop(backend::REPL.REPLBackend)\n    @ REPL ~/.asdf/installs/julia/1.7.0/share/julia/stdlib/v1.7/REPL/src/REPL.jl:244\n  [8] start_repl_backend(backend::REPL.REPLBackend, consumer::Any)\n    @ REPL ~/.asdf/installs/julia/1.7.0/share/julia/stdlib/v1.7/REPL/src/REPL.jl:229\n  [9] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool)\n    @ REPL ~/.asdf/installs/julia/1.7.0/share/julia/stdlib/v1.7/REPL/src/REPL.jl:362\n [10] run_repl(repl::REPL.AbstractREPL, consumer::Any)\n    @ REPL ~/.asdf/installs/julia/1.7.0/share/julia/stdlib/v1.7/REPL/src/REPL.jl:349\n [11] (::Base.var\"#930#932\"{Bool, Bool, Bool})(REPL::Module)\n    @ Base ./client.jl:394\n [12] #invokelatest#2\n    @ ./essentials.jl:716 [inlined]\n [13] invokelatest\n    @ ./essentials.jl:714 [inlined]\n [14] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)\n    @ Base ./client.jl:379\n [15] exec_options(opts::Base.JLOptions)\n    @ Base ./client.jl:309\n [16] _start()\n    @ Base ./client.jl:495"]}}
```
Comment (fredrikekre) [inline README.md]: What do you think about scrubbing the backtrace? https://julialang.zulipchat.com/#narrow/stream/274208-helpdesk-.28published.29/topic/filtering.20backtraces

Comment (ericphanson) [reply README.md]: I tried `Base.display_error` that does the scrubbing for us, what do you think?

File: src/LoggingFormats.jl [hunk 2]
```diff
@@ -70,6 +70,17 @@ end
 transform(::Type{String}, v) = string(v)
 transform(::Type{Any}, v) = v
 
+# Use key information, then lower to 2-arg transform
+function transform(::Type{T}, key, v) where {T}
+    key == :exception || return transform(T, v)
+    if v isa Tuple && length(v) == 2
```
Comment (fredrikekre) [inline src/LoggingFormats.jl]: Maybe make sure that it is also an exception?
```suggestion
    if v isa Tuple && length(v) == 2 && v[1] isa Exception
```

File: src/LoggingFormats.jl [hunk 3]
```diff
@@ -70,6 +70,17 @@ end
...
+        e, bt = v
+        msg = sprint(Base.display_error, e, bt)
+        return transform(T, (string(e), msg))
```
Comment (fredrikekre) [inline src/LoggingFormats.jl]: Why the tuple and not just `msg`?

Comment (ericphanson) [reply src/LoggingFormats.jl]:
```suggestion
        return transform(T, msg)
```

**Comment by ericphanson** [comment]
with recursive JSON we could do something like emit a list of stackframes and maybe you could do some analysis to see what lines are showing up the most or something over a large codebase... But probably that's generally not going to be used, and you can always just use a different key like `stacktrace = jsonify(stacktrace(catch_backtrace()))` or something if you do want that (for some appropriate `jsonify`).

**Pull request merged**